workers/datasets_based/src/datasets_based/workers/first_rows.py

```diff
@@ -480,6 +506,24 @@ def compute_first_rows_response(
             raise FeaturesError("The split features (columns) cannot be extracted.", cause=err) from err
     else:
         features = info.features
+
+    if len(features) > columns_max_number:
```

Comment:
beware, features can be None (as for the types). We have to check this before calling len()

```suggestion
    if features and len(features) > columns_max_number:
```